### PR TITLE
feat(certificate): dedicated color for expiration date

### PR DESCRIPTION
**Pull request by stonebuzz**
Dedicated color compute for ```Certificate``` ```expiration_date```

## If notification is disabled, use standard compute 

```warning``` if ```expiration_date``` ```<``` ```date now```

## If notification is enabled

image
 

```red``` if ```expiration_date``` ```<``` ```date now```

```warning``` if ```expiration_date``` ```<``` ```date now + before ```

else ```green``` 

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !25253

### CODE REVIEW

**Comment by orthagh** [comment]
Can you post a screenshot of the final render in Search/Form/etc ?

**Comment by stonebuzz** [comment]
image

image

image

image

image

image

File: css/legacy/includes/_styles.scss.orig [hunk 1]
```diff
@@ -0,0 +1,1539 @@
+/*!
```
[the remark this answers is not in the document]
Comment (trasher) [reply css/legacy/includes/_styles.scss.orig:1]: Search.php.orig neither

[the remark this answers is not in the document]
Comment (Rom1-B) [reply css/legacy/includes/_styles.scss.orig:1]: `generic_show_form.html.twig.orig` and `generic_show_form.html.twig.rej`

File: css/legacy/includes/_styles.scss [hunk 2]
```diff
@@ -1536,4 +1536,4 @@ div.banner-need-update {
    font-weight: bold;
    padding: 10px;
    text-align: right;
-}
+}
```
Comment (trasher) [inline css/legacy/includes/_styles.scss:1539]: does not seems required

File: src/Search.php [hunk 3]
```diff
@@ -6211,7 +6211,24 @@ public static function displayConfigItem($itemtype, $ID, $data = [])
...
+                ) {
+                    $out = "";
+                    if ($before = Entity::getUsedConfig('send_certificates_alert_before_delay', $_SESSION['glpiactive_entity'])) {
+                        $before = date('Y-m-d', strtotime($_SESSION['glpi_currenttime'] . " + $before days"));
+                        if ($data[$NAME][0]['name'] < $_SESSION['glpi_currenttime']) {
+                            $out = " class=\"shadow-none\" style=\"background-color: #d63939\" ";
+                        } elseif ($data[$NAME][0]['name'] < $before) {
+                            $out = " class=\"shadow-none\"  style=\"background-color: #de5d06\" ";
+                        } elseif ($data[$NAME][0]['name'] >= $before) {
+                            $out = " class=\"shadow-none\"  style=\"background-color: #a1cf66\" ";
```
Comment (Rom1-B) [inline src/Search.php:6227]: Black on red is not very readable, maybe white should be used?

Comment (stonebuzz) [reply src/Search.php]: image

File: src/Search.php [hunk 4]
```diff
@@ -6211,7 +6211,24 @@ public static function displayConfigItem($itemtype, $ID, $data = [])
                     $out = " class=\"shadow-none\" style=\"background-color: #cf9b9b\" ";
                 }
                 break;
-
+            case "glpi_certificates.date_expiration":
+                if (
```
Comment (Rom1-B) [inline src/Search.php:6215]: `"glpi_certificates.date_expiration":` was already in the function (at the end), so now it is there twice

Comment (stonebuzz) [reply src/Search.php:6215]: refactored

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by Rom1-B — APPROVED** [review]

**Pull request merged**